domain/application/state/application.go

```diff
@@ -1874,3 +1874,78 @@ func decodeArchitecture(arch int) (application.Architecture, error) {
 		return -1, fmt.Errorf("unsupported architecture: %d", arch)
 	}
 }
+
+// GetApplicationIDByUnitName returns the application ID for the named unit,
+// returning an error satisfying [applicationerrors.UnitNotFound] if the unit
+// doesn't exist.
+func (st *ApplicationState) GetApplicationIDByUnitName(
+	ctx context.Context,
+	name coreunit.Name,
+) (coreapplication.ID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Trace(err)
```

Comment:
**todo:** During the sprint we decide to move out from `juju/errors` in favor of `internal/juju/errors`. 

In [this commit](https://github.com/juju/juju/commit/f3c98a5943c4d2abacfedd0f7db3b8580ceaef59), `Capture` has been introduced to provide the same kind of functionalities than `Trace`

even if we don't want to update now the whole file, we should at least use `internalerrors.Capture` in new code